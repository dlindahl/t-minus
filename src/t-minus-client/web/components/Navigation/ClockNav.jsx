import AppNavBar from './AppNavBar'
import Colors from '../../../shared/constants/Colors'
import ResetButton from '../Buttons/ResetButton'
import StartStopButton from '../Buttons/StartStopButton'
import StopwatchModeToggleButton from '../Buttons/StopwatchModeToggleButton'
import TimerModeToggleButton from '../Buttons/TimerModeToggleButton'
import Title from 'react-icons/lib/md/title'

const ClockNav = (props) => (
  <AppNavBar direction="column">
    <Title color={Colors.Clouds} size={50}/>
    <TimerModeToggleButton/>
    <StopwatchModeToggleButton/>
    <StartStopButton/>
    &nbsp;
    <ResetButton/>
  </AppNavBar>
)

export default ClockNav
