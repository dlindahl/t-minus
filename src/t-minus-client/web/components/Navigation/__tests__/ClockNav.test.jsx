import ClockNav from '../ClockNav'
import configureStore from 'redux-mock-store'
import emptyObj from 'empty/object'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

const STATES = {
  default: {
    clock: emptyObj,
    display: emptyObj,
    meta: emptyObj
  }
}
const store = configureStore()

describe('ClockNav', () => {
  it('renders the component', () => {
    const component = renderer.create(
      <Provider store={store(STATES.default)}>
        <ClockNav/>
      </Provider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
