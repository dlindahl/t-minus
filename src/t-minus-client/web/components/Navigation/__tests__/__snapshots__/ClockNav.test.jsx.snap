exports[`ClockNav renders the component 1`] = `
<nav
  style={
    Object {
      "background": "#2c3e50",
      "width": 50,
    }
  }>
  <ul
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "listStyleType": "none",
        "margin": 0,
        "padding": 0,
      }
    }>
    <li
      style={
        Object {
          "textAlign": "center",
        }
      }>
      <svg
        fill="currentColor"
        height={50}
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": "#ecf0f1",
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width={50}>
        <g>
          <path
            d="m8.4 6.6h23.2v5h-9.1v20h-5v-20h-9.1v-5z" />
        </g>
      </svg>
    </li>
    <li
      style={
        Object {
          "textAlign": "center",
        }
      }>
      <button
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "background": "transparent",
            "border": 0,
            "color": "#bdc3c7",
            "cursor": "pointer",
            "fontSize": "2rem",
            "height": 50,
            "lineHeight": 1,
            "padding": 0,
            "width": 50,
          }
        }>
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "top",
            }
          }
          viewBox="0 0 40 40"
          width="1em">
          <g>
            <path
              d="m20 33.4q4.8 0 8.2-3.5t3.4-8.3-3.4-8.2-8.2-3.4-8.2 3.4-3.4 8.2 3.4 8.3 8.2 3.5z m0-26.8q6.3 0 10.6 4.5t4.4 10.5-4.4 10.6-10.6 4.4-10.6-4.4-4.4-10.6 4.4-10.5 10.6-4.5z m0.9 6.8v8.7l6.6 3.9-1.2 2.1-7.9-4.7v-10h2.5z m-7.8-7.8l-7.6 6.4-2.1-2.5 7.6-6.4z m23.5 3.9l-2.1 2.6-7.6-6.5 2.1-2.5z" />
          </g>
        </svg>
      </button>
    </li>
    <li
      style={
        Object {
          "textAlign": "center",
        }
      }>
      <button
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "background": "transparent",
            "border": 0,
            "color": "#bdc3c7",
            "cursor": "pointer",
            "fontSize": "2rem",
            "height": 50,
            "lineHeight": 1,
            "padding": 0,
            "width": 50,
          }
        }>
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "top",
            }
          }
          viewBox="0 0 40 40"
          width="1em">
          <g>
            <path
              d="m20.9 11.6v8.8l7.5 4.4-1.3 2.2-8.7-5.4v-10h2.5z m-0.9 21.8c7.3 0 13.4-6.1 13.4-13.4s-6.1-13.4-13.4-13.4-13.4 6.1-13.4 13.4 6.1 13.4 13.4 13.4z m0-30c9.2 0 16.6 7.4 16.6 16.6s-7.4 16.6-16.6 16.6-16.6-7.4-16.6-16.6 7.4-16.6 16.6-16.6z" />
          </g>
        </svg>
      </button>
    </li>
    <li
      style={
        Object {
          "textAlign": "center",
        }
      }>
      <button
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "background": "transparent",
            "border": 0,
            "color": "#bdc3c7",
            "cursor": "pointer",
            "fontSize": "2rem",
            "height": 50,
            "lineHeight": 1,
            "padding": 0,
            "width": 50,
          }
        }>
        <svg
          data-test-id="PlayArrow"
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "top",
            }
          }
          viewBox="0 0 40 40"
          width="1em">
          <g>
            <path
              d="m13.4 8.4l18.2 11.6-18.2 11.6v-23.2z" />
          </g>
        </svg>
      </button>
    </li>
    <li
      style={
        Object {
          "flex": 1,
        }
      }>
      Â 
    </li>
    <li
      style={
        Object {
          "textAlign": "center",
        }
      }>
      <button
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "background": "transparent",
            "border": 0,
            "color": "#bdc3c7",
            "cursor": "pointer",
            "fontSize": "2rem",
            "height": 50,
            "lineHeight": 1,
            "padding": 0,
            "width": 50,
          }
        }>
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "top",
            }
          }
          viewBox="0 0 40 40"
          width="1em">
          <g>
            <path
              d="m31.3 12.9c1.3 2 2.1 4.5 2.1 7.1 0 7.3-6.1 13.4-13.4 13.4v5l-6.6-6.8 6.6-6.6v5c5.5 0 10-4.4 10-10 0-1.7-0.5-3.3-1.2-4.7z m-11.3-2.9c-5.5 0-10 4.5-10 10 0 1.7 0.4 3.3 1.2 4.7l-2.4 2.4c-1.4-2-2.2-4.5-2.2-7.1 0-7.3 6.1-13.4 13.4-13.4v-5l6.6 6.8-6.6 6.6v-5z" />
          </g>
        </svg>
      </button>
    </li>
  </ul>
</nav>
`;
